src/pymor/reductors/mt.py

```diff
@@ -0,0 +1,237 @@
+            - `'LR'`: select eigenvalues with largest real part
+            - `'NR'`: select eigenvalues with largest norm(residual) / abs(Re(pole))
+            - `'NS'`: select eigenvalues with largest norm(residual) / abs(pole)
+            - `'NM'`: select eigenvalues with largest norm(residual)
+        method_options
+            Optional dict with more options for the samdp algorithm.
+        take_complex
+            If `True` the reduced model is complex when the poles of the reduced
+            model are not closed under complex conjugation.
+
+        Returns
+        -------
+        rom
+            Reduced-order model.
+        """
+        assert 0 < r < self.fom.order
+        assert projection in ('orth', 'biorth')
+        assert decomposition in ('eig', 'samdp')
+        assert which in ('LR', 'SM', 'NR', 'NS', 'NM')
+        assert method_options is None or isinstance(method_options, dict)
+        if not method_options:
+            method_options = {'which':'LR'}
+
+        if self.fom.parametric:
+            fom = self.fom.with_(**{op: getattr(self.fom, op).assemble(mu=self.mu)
+                                    for op in ['A', 'B', 'C', 'D', 'E']})
+        else:
+            fom = self.fom
+
+        self.V = fom.A.source.empty(reserve=r)
+        self.W = fom.A.source.empty(reserve=r)
+
+        if decomposition == 'eig':
+            if fom.order >= sparse_min_size():
+                if not isinstance(fom.A, NumpyMatrixOperator) or fom.A.sparse:
+                    self.logger.warning('Converting operator A to a NumPy array.')
+                if not isinstance(fom.E, IdentityOperator):
+                    if not isinstance(fom.E, NumpyMatrixOperator) or fom.E.sparse:
+                        self.logger.warning('Converting operator E to a NumPy array.')
+            A = to_matrix(fom.A, format='dense')
+            E = None if isinstance(fom.E, IdentityOperator) else to_matrix(fom.E, format='dense')
+
+            if symmetric:
+                poles, ev_r = spla.eig(A, E, right=True)
+                rev = fom.A.source.from_numpy(ev_r.T)
+                lev = rev.copy()
+            else:
+                poles, ev_l, ev_r = spla.eig(A, E, left=True, right=True)
+                rev = fom.A.source.from_numpy(ev_r.T)
+                lev = fom.A.source.from_numpy(ev_l.T)
+            if which == 'SM':
+                idx = np.argsort(np.abs(poles))
+                dominance = np.abs(poles)
+            elif which == 'LR':
+                idx = np.argsort(-poles.real)
+                dominance = np.ones(len(poles))
+            else:
+                absres = np.empty(len(poles))
+                for i in range(len(poles)):
+                    lev[i].scal(1 / lev[i].inner(fom.E.apply(rev[i]))[0][0])
+                    b_norm = fom.B.apply_adjoint(lev[i]).norm()
+                    c_norm = fom.C.apply(rev[i]).norm()
+                    absres[i] = b_norm @ c_norm
```

Comment:
Using `@` for 1x1 matrices seems a bit strange.
```suggestion
                    b_norm = fom.B.apply_adjoint(lev[i]).norm()[0]
                    c_norm = fom.C.apply(rev[i]).norm()[0]
                    absres[i] = b_norm * c_norm
```